saleor/graphql/product/mutations/products.py

```diff
@@ -979,20 +985,31 @@ def success_response(cls, instance):
     @classmethod
     def perform_mutation(cls, _root, info, **data):
         node_id = data.get("id")
-        variant_pk = from_global_id_strict_type(node_id, ProductVariant, field="pk")
+        instance = cls.get_node_or_error(info, node_id, only_type=ProductVariant)
 
         # get draft order lines for variant
         line_pks = list(
             order_models.OrderLine.objects.filter(
-                variant__pk=variant_pk, order__status=OrderStatus.DRAFT
+                variant__pk=instance.pk, order__status=OrderStatus.DRAFT
             ).values_list("pk", flat=True)
         )
 
+        # Get cached variant with related fields to fully populate webhook payload.
+        variant = (
+            models.ProductVariant.objects.prefetch_related(
+                "channel_listings",
+                "attributes__values",
+            )
+        ).get(id=instance.id)
+        with transaction.atomic():
```

Comment:
Hhm... It is strange. You're applying the transaction only to call transaction.on_commit 😄 
Also to be consistent with `product_deleted` event. You should call the event that variant was removed after we really remove it from DB.